### ISSUE TITLE
Hovercard popovers persist after parent Hovercard component dismount

### ISSUE THREAD

**Issue by Addison-Stavlo** [Automattic/gravatar#210]
## Issue
When a Gravatar with Hovercards is used as a navigation link, the hovercard popover remains visible after navigation occurs and the corresponding Hovercard component dismounts.

## Current Behavior
1. User hovers over a Gravatar, hovercard appears
2. User clicks the Gravatar to navigate to a new page
3. The Hovercards component unmounts
4. The hovercard popover remains visible on the new page

A gif example of this happening within the WordPress Reader (in calypso) can be found here - https://github.com/Automattic/wp-calypso/pull/104136#discussion_r2140657044 

## Expected Behavior
The hovercard should be automatically cleaned up when the Hovercards component unmounts, regardless of how the unmounting occurs (navigation, state changes, etc.).

## Workaround
We've implemented a workaround in our application by manually cleaning up the hovercard elements on component unmount:

```jsx
useEffect(() => {
  return () => {
    const hovercards = document.querySelectorAll('.gravatar-hovercard');
    hovercards.forEach(card => card.remove());
  };
}, []);
```

EDIT - This does not fully work, with delays for popover appearance there are still circumstances where the popover will appear after the component has unmounted and already triggered this hacky cleanup. Extending the fix required setting a reference on a parent container, and evaluating if that parent container is still present on the dom during `onCanShowHovercard`

## Suggested Fix
The Hovercards component should handle this cleanup internally, perhaps by:
1. Tracking any hovercard elements it creates
2. Cleaning up these elements in its own unmount logic
3. Using a cleanup function in useEffect to ensure proper cleanup

## Additional Context
- This issue is particularly noticeable in single-page applications where navigation doesn't cause a full page reload
- While a workaround is available, this should generally be handled by the component library rather than requiring workarounds in consuming applications.

## Environment
- @gravatar-com/hovercards version: 0.13.0
- React version: 18.3.1

**Comment by wellyshen** [comment]
@Addison-Stavlo Thank you for reporting this issue. I think we need an API method for cleaning up hovercards, and also cleaning up hovercards in [this detach method](https://github.com/Automattic/gravatar/tree/trunk/web/packages/hovercards#detach-void).

**Comment by wellyshen** [comment]
@Addison-Stavlo, FYI. [This PR](https://github.com/Automattic/gravatar/pull/216) adds a new `hideOnTargetClick` option and updates the `detach()` to remove hovercards.

Once the new version is released, choose whichever option best suits your needs.

```tsx
const hovercards = new Hovercards( { hideOnTargetClick: true } );

// Or

useEffect(() => {
  return () => hovercards.detach();
}, []);
```

**Comment by wellyshen** [comment]
@Addison-Stavlo [The new APIs have been released.](https://github.com/Automattic/gravatar/releases/tag/hovercards-0.14.0)

**Comment by Addison-Stavlo** [comment]
Very nice, we may end up using both. Thank you!
